python-sdk/src/transfers/datasets/base.py

```diff
@@ -0,0 +1,24 @@
+from __future__ import annotations
+
+from airflow.utils.log.logging_mixin import LoggingMixin
+from attr import define, field
+
+from astro.airflow.datasets import Dataset
+
+
+@define
+class UniversalDataset(LoggingMixin, Dataset):
```

Comment:
hmm, why do we need a `UniversalDataset`

Reply:
File, Table, Dataframe and API classes have some common method which would be part of `UniversalDataset`: https://github.com/astronomer/astro-sdk/pull/1267/commits/f36c537402cb4bb26df2aa279698882208105b5e